src/Service/PackageSynchronizer/ComposerPackageSynchronizer.php

```diff
@@ -79,6 +79,13 @@ public function synchronize(Package $package): void
 
             $latest = current($packages);
 
+            foreach ($packages as $p) {
+                if ($p->getStability() === Version::STABILITY_STABLE) {
```

Comment:
we can put this `if` inside existing loop, also please provided test for that case